app/gui2/src/components/visualizations/TableVisualization.vue

```diff
@@ -265,12 +270,40 @@ function isMatrix(data: object): data is LegacyMatrix {
   return json.every((d) => d.length === firstLen)
 }
 
-function toField(name: string): ColDef {
-  return { field: name }
+function toField(name: string, valType?: string): ColDef {
```

Comment:
Let's add an explicit `valType?: string | undefined | null` here, since you do pass an explicit null there in one occasion.